cmd/kubeadm/app/cmd/token.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"text/tabwriter"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubemaster "k8s.io/kubernetes/cmd/kubeadm/app/master"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/kubectl"
+)
+
+const (
+	bootstrapTokenSecretPrefix = "bootstrap-token-"
+	defaultTokenDuration       = time.Duration(8) * time.Hour
+)
+
+func NewCmdToken(out io.Writer) *cobra.Command {
+	tokenCmd := &cobra.Command{
+		Use:   "token",
+		Short: "Manage bootstrap tokens",
+	}
+
+	var tokenDuration time.Duration
+	tokenSecret := &kubeadmapi.Secrets{}
+	createCmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create discovery tokens on the server.",
+		Run: func(tokenCmd *cobra.Command, args []string) {
+			err := RunCreateToken(out, tokenCmd, tokenDuration, tokenSecret)
+			kubeadmutil.CheckErr(err)
+		},
+	}
+	createCmd.PersistentFlags().DurationVar(&tokenDuration,
+		"duration", defaultTokenDuration, "Duration the token should be valid")
```

Comment:
TTL sounds good. 

Mentioning no TTL feels a little worrisome, do you feel that's important to expose if users want to opt into it? Or could we get away with just not offering an unlimited bootstrap token.